src/pudl/transform/ferc1.py

```diff
@@ -1021,8 +1293,500 @@ def process_xbrl_metadata(self, xbrl_metadata_json) -> pd.DataFrame:
                     "row_type_xbrl": pd.StringDtype(),
                 }
             )
+            # Everything below here deals with correcting the calculations.
+            .assign(
+                xbrl_factoid_name_original=lambda x: x.xbrl_factoid,
+                inter_table_calc_flag=lambda x: x.calculations.str.contains(
+                    "source_table"
+                ),
+            )
+        )
+        xbrl_factoid_name_map = {
+            xbrl_factoid_name_og: self.rename_column_xbrl_to_pudl(xbrl_factoid_name_og)
+            for xbrl_factoid_name_og in tbl_meta.xbrl_factoid
+        }
+        tbl_meta.xbrl_factoid = tbl_meta.xbrl_factoid.map(xbrl_factoid_name_map)
+
+        def rename_calculation_components(calc: str) -> str:
+            # apply the rename for the "name" element of all of the calc components
+            renamed_calc = [
+                {
+                    k: self.rename_column_xbrl_to_pudl(v) if k == "name" else v
+                    for (k, v) in calc_component.items()
+                }
+                for calc_component in json.loads(calc)
+            ]
+            return json.dumps(renamed_calc)
+
+        tbl_meta.calculations = tbl_meta.calculations.apply(
+            rename_calculation_components
+        )
+        tbl_meta = (
+            self.apply_xbrl_calculation_fixes(tbl_meta)
+            .pipe(self.remove_duplicated_calculation_components)
+            .pipe(self.add_calculation_correction_components)
+        )
+        tbl_meta.calculations = tbl_meta.calculations.astype(pd.StringDtype())
+
+        return tbl_meta
+
+    def rename_column_xbrl_to_pudl(
```

Comment:
The name of this method sounds like it's actually going to rename some columns, but it's just generating a mapping from old to new column names right? Would it be clearer as a @property called xbrl_to_pudl_column_map?

Reply:
this method takes *one* column name from the xbrl metdata and converts it to the end-state pudl name. the map you are trying to name here is built inside of `process_xbrl_metadata` using this method on each raw factoid name (its called `xbrl_factoid_name_map` up there).

I think it would take some cartwheeling to make the map into a stand alone property. Because we need to use this map within `process_xbrl_metadata`, but also in order to make the map, we need the "raw" factoid names, which we only have halfway through `process_xbrl_metadata`.

But i hear you that this maybe isn't the right name for this method. its not even a column (well the factoids kinda are column names in xbrl but not once we've wide-to-tidy-ed them). how about `rename_raw_xbrl_factoid_to_pudl_name`?

Reply:
rename the rename to `raw_xbrl_factoid_to_pudl_name`